integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/v2/PeekNextMoveMessageInImpl.java

```diff
@@ -0,0 +1,99 @@
+package no.difi.meldingsutveksling.nextmove.v2;
+
+import lombok.extern.slf4j.Slf4j;
+import no.difi.meldingsutveksling.nextmove.NextMoveInMessage;
+import org.hibernate.CacheMode;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.persistence.EntityManager;
+import javax.persistence.LockModeType;
+import javax.persistence.PersistenceContext;
+import javax.persistence.PessimisticLockException;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import java.time.OffsetDateTime;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+public class PeekNextMoveMessageInImpl implements PeekNextMoveMessageIn {
+
+    @PersistenceContext
+    private EntityManager em;
+
+    @Override
+    @Transactional(readOnly = true, isolation = Isolation.READ_UNCOMMITTED)
+    public List<Long> peek(NextMoveInMessageQueryInput input, int maxResults) {
+        try {
```

Comment:
Gjerne ein liten overordna kommentar om peek-strategien, her eller anna plass.

- EntityManager direkte for å unngå diverse caching og locking i Spring data
- READ_UNCOMMITTED og cache disabled for at lock skal reflekteres i uthenta id-er så kjapt som mulig (og ikkje vente på heile lock-transaksjonen)
- Hente opp til 20 kandidater for å tåle nokre kollisjoner på lock utan å måtte hente fleire fra database
- Loop gjennom kandidater til ein fakisk får LOCK ok
- ber i praksis konsument om å vente litt mha å returnere NO_CONTENT dersom ein ikkje får LOCK på nokon av kandidatane